### PR TITLE
[v12] SAML identity provider docs.

### PR DESCRIPTION
**Pull request by mdwn**
Backporting https://github.com/gravitational/teleport/pull/22414 to branch/v12

### CODE REVIEW

**Review by r0mant — APPROVED** [review]

**Review by ptgott — APPROVED** [review]

File: docs/pages/access-controls/idps.mdx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+title: Configure Teleport as an identity provider
+description: How to set up Teleport's identity provider functionality
+---
+
+Since Teleport 12.1 users can authenticate to both internal and external applications
+through the use of a built in identity provider.
```
Comment (alexfornuto) [inline docs/pages/access-controls/idps.mdx:7]: Should we mention that it's in preview mode on this page as well?

Comment (mdwn) [reply docs/pages/access-controls/idps.mdx:7]: Ahhh yes, good move. Will do!

**Review by alexfornuto — APPROVED** [review]

File: docs/config.json [hunk 2]
```diff
@@ -312,6 +312,28 @@
             }
           ]
         },
+        {
+          "title": "Teleport as an IdP",
```
Comment (r0mant) [inline docs/config.json:316]:
```suggestion
          "title": "Teleport as an IdP (Preview)",
```

**Review by r0mant — COMMENTED** [review]
@mdwn Can you also add "preview" note to the ToC entry title?

**Pull request merged**